packages/web/utils/total-value.ts

```diff
@@ -0,0 +1,36 @@
+import { FiatCurrency } from "@keplr-wallet/types";
+import { CoinPretty, PricePretty } from "@keplr-wallet/unit";
+import { PoolFallbackPriceStore } from "@osmosis-labs/stores";
+
+// TODO: Change this out of the price pretty overhead, in a later iteration.
+export function coinsToFiatValue(
+  priceStore: PoolFallbackPriceStore,
+  fiat: FiatCurrency,
+  coins: CoinPretty[]
+): number {
+  return Number(
+    coinsToFiatPricePretty(priceStore, fiat, coins).toDec().toString()
+  );
+}
+
+// TODO: Do we actually want this anywhere?
+export function coinsToFiatPricePretty(
+  priceStore: PoolFallbackPriceStore,
```

Comment:
IMO we'd get more type flexibility if we used `IPriceStore` instead of the concrete `PoolFallbackPriceStore` that we have.

Stepping back, I think we'd make good use of a similar function on a new `IPriceStore` that we could add to our impl at `PoolFallbackPriceStore` to take a list of coins and return a single price. Rather than have these free floating util functions. I think it may be more discoverable and of higher locality. Thoughts?

Stepping all the way back and longer term, and I've brought this up with tony and Josh. I think we should extract the keplr-wallet architecture into a cosmos-eco maintained set of packages where the price store mentioned above could include this functionality for the greater benefit of the Cosmos frontend ecosystem. I think the keplr framework we're using has offered tons of value and has been a great interchain foundation to build from but it should be decoupled from Keplr team's initiatives and maintained and branded separately. cc @pyramation cc @sunnya97

Reply:
* Will change the concrete price store struct to the interface
* I don't quite understand our stores abstractions, so I don't have an opinion. I generally prefer just getting everything into one function first that is obviously state compatible, and then in a second pass moving how we like longer term. Happy to go with whatever, until I get a better sense of how we use stores.


Pretty sure theres universal dependence to get off of keplr's type system for this stuff :)

e.g. In the relevant code, we should move to just floats. 
What I was hoping to do here was:
* In a later PR, get tests for the unit functions, and then actually change the internal business accounting to floats, and return a final price pretty.
* Have a discussion later on for a single function for going from PricePretty (or whatever "price type" we use) to display, so we don't do ad-hoc ToDec().Stirng()'s, and manually set precisions around the codebase.

Reply:
We should definitely get on a call to discuss this stuff if you consider it high priority, as I think there's a lot of background on the state of the frontend